homeassistant/components/sensor/buienradar.py

```diff
@@ -0,0 +1,293 @@
+    """Setup the buienradar_nl sensor."""
+    latitude = config.get(CONF_LATITUDE, hass.config.latitude)
+    longitude = config.get(CONF_LONGITUDE, hass.config.longitude)
+
+    if None in (latitude, longitude):
+        _LOGGER.error("Latitude or longitude not set in HomeAssistant config")
+        return False
+
+    coordinates = {CONF_LATITUDE: float(latitude),
+                   CONF_LONGITUDE: float(longitude)}
+
+    dev = []
+    for sensor_type in config[CONF_MONITORED_CONDITIONS]:
+        dev.append(BrSensor(sensor_type))
+    async_add_devices(dev)
+
+    data = BrData(hass, coordinates, dev)
+    # schedule the first update in 1 minute from now:
+    data.schedule_update(1)
+
+
+class BrSensor(Entity):
+    """Representation of an Buienradar sensor."""
+
+    def __init__(self, sensor_type):
+        """Initialize the sensor."""
+        self.client_name = 'br'
+        self._name = SENSOR_TYPES[sensor_type][0]
+        self.type = sensor_type
+        self._state = None
+        self._unit_of_measurement = SENSOR_TYPES[self.type][1]
+        self._entity_picture = None
+        self._attribution = None
+
+    def load_data(self, data):
+        """Load the sensor with relevant data."""
+        # Find sensor
+        from buienradar.buienradar import (ATTRIBUTION, IMAGE, SYMBOL)
+
+        self._attribution = data.get(ATTRIBUTION)
+        if self.type == SYMBOL:
+            # update weather symbol & status text
+            new_state = data.get(self.type)
+            img = data.get(IMAGE)
+
+            # pylint: disable=protected-access
+            if new_state != self._state or img != self._entity_picture:
+                self._state = new_state
+                self._entity_picture = img
+                return True
+        else:
+            # update all other sensors
+            new_state = data.get(self.type)
+            # pylint: disable=protected-access
+            if new_state != self._state:
+                self._state = new_state
+                return True
+
+    @property
+    def attribution(self):
+        """Return the attribution."""
+        return self._attribution
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return '{} {}'.format(self.client_name, self._name)
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self._state
+
+    @property
+    def should_poll(self):  # pylint: disable=no-self-use
+        """No polling needed."""
+        return False
+
+    @property
+    def entity_picture(self):
+        """Weather symbol if type is symbol."""
+        from buienradar.buienradar import SYMBOL
+
+        if self.type != SYMBOL:
+            return None
+        else:
+            return self._entity_picture
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        return {
+            ATTR_ATTRIBUTION: self._attribution,
```

Comment:
Could be useful to add the station name to the attributes and not returning it as its own sensor. If you run with multiple stations then you could see where the value is coming from.